UM/Settings/SettingFunction.py

```diff
@@ -194,39 +194,88 @@ def visit_Name(self, node: ast.Name) -> None:  # [CodeStyle: ast.NodeVisitor req
             self.values.add(node.id)
             self.keys.add(node.id)
 
+    def visit_DictComp(self, node: ast.DictComp) -> None:
+        # This is mostly because there is no reason to use it at the moment. It might also be an attack vector.
+        raise IllegalMethodError("Dict Comprehension is not allowed")
+
+    def visit_ListComp(self, node):
+        # This is mostly because there is no reason to use it at the moment. It might also be an attack vector.
+        raise IllegalMethodError("List Comprehension is not allowed")
+
+    def visit_Attribute(self, node: ast.Attribute) -> None:
+        if node.attr not in self._knownNames:
+            raise IllegalMethodError(node.attr)
+        self.visit(node.value)  # Go a step deeper
+
+    def generic_visit(self, node):
+        for child_node in ast.iter_child_nodes(node):
+            self.visit(child_node)
+
     def visit_Str(self, node: ast.AST) -> None:
         """This one is used before Python 3.8 to visit string types.
         
         visit_Str will be marked as deprecated from Python 3.8 and onwards.
         """
+        # The blacklisting is done just in case (All function calls should be whitelisted. The blacklist is to make
+        # extra sure that certain calls are *not* made!)
+        if node.s in self._blacklist:
+            raise IllegalMethodError(node.s)
```

Comment:
You are blacklisting only constant values.

Shouldn't you also blacklist similarly in `visit_Call` ?
It seems to me we should check all of the possible [classnames](See https://docs.python.org/3/library/ast.html#node-classes) and override as many as possible using `visit_<classname>`.

i.e. ExtSlice, SetComp, Starred, etc?

Reply:
GeneratorExp?

Reply:
You can't blacklist call like that, because it doesn't have an ID to check for.